staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restclient
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/tools/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/component-base/metrics/testutil"
+)
+
+func TestClientGOMetrics(t *testing.T) {
+	tests := []struct {
+		description string
+		name        string
+		metric      interface{}
+		update      func()
+		want        string
+	}{
+		{
+			description: "Number of HTTP requests, partitioned by status code, method, and host.",
+			name:        "rest_client_requests_total",
+			metric:      requestResult,
+			update: func() {
+				metrics.RequestResult.Increment(context.TODO(), "200", "POST", "www.foo.com")
+			},
+			want: `
+			            # HELP rest_client_requests_total [ALPHA] Number of HTTP requests, partitioned by status code, method, and host.
+			            # TYPE rest_client_requests_total counter
+			            rest_client_requests_total{code="200",host="www.foo.com",method="POST"} 1
```

Comment:
```suggestion
			            rest_client_requests_total{code="200",host="www.foo.com",verb="POST"} 1
```
@tkashem this needs to be updated to fix the test failure

Reply:
hmm, interesting. this PR makes no change in the `requests_total` metric
https://github.com/kubernetes/kubernetes/pull/108396/files#diff-426208a8788717317d1215b826579c69423d2f8586f9d41529d77f403541342aR76-R83

but the failure seems to be coming from `rest_client_requests_total` test
```
=== FAIL: vendor/k8s.io/component-base/metrics/prometheus/restclient TestClientGOMetrics/Number_of_HTTP_requests,_partitioned_by_status_code,_method,_and_host. (0.03s)
    metrics_test.go:83: 
        
        Diff:
        --- metric output does not match expectation; want
        +++ got:
        @@ -1,4 +1,4 @@
        -(bytes.Buffer) # HELP rest_client_requests_total [ALPHA] Number of HTTP requests, partitioned by status code, method, and host.
        +(bytes.Buffer) # HELP rest_client_requests_total [ALPHA] Number of HTTP requests, partitioned by status code, verb, and host.
         # TYPE rest_client_requests_total counter
        -rest_client_requests_total{code="200",host="www.foo.com",method="POST"} 1
        +rest_client_requests_total{code="200",host="www.foo.com",verb="POST"} 1
         
        
    --- FAIL: TestClientGOMetrics/Number_of_HTTP_requests,_partitioned_by_status_code,_method,_and_host. (0.03s)
```

am i missing something here @dgrisonnet? 


I ran `stress` but could not reproduce it yet
```
$ stress ./restclient.test -test.run=TestClientGOMetrics
...
4m40s: 33098 runs so far, 0 failures
```

Reply:
Sorry for not adding more context, it broke because of my recent change in https://github.com/kubernetes/kubernetes/pull/115455.

Reply:
Rebasing + updating the new test should do it

Reply:
that makes sense, thanks!